<launch> 

  <arg name="vis" default="true" doc="Launch RViZ"/>
  <arg name="source" default="realsense" doc="Describes the camera we are using"/>
  

  <arg name="x" default="0.89535" doc="Description"/>
  <arg name="y" default="-0.88265" doc="Description"/>
  <arg name="z" default="2.16535" doc="Description"/>


  <arg name="qx" default="1.0" doc="Description"/>
  <arg name="qy" default="0.0" doc="Description"/>
  <arg name="qz" default="0" doc="Description"/>
  <arg name="qw" default="0" doc="Description"/>



  <!-- static_transform_publisher x y z qx qy qz qw frame_id child_frame_id  period_in_ms -->
  <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="$(arg x) $(arg y) $(arg z) $(arg qx) $(arg qy) 
	$(arg qz) $(arg qw)  base camera_depth_frame 100" />
  
  <!-- intel realsense, which publishes on /camera/depth/color/points instead of /camera/depth/points -->
  <remap from="/camera/depth/color/points" to="/camera/depth/points" />

<include if="$(eval source == 'realsense')" file="$(find realsense2_camera)/launch/rs_camera.launch" >
       <arg name="filters" value="pointcloud" />
  </include>

  <!-- launch the nodelet manager -->
  <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen"/>  

  <!-- pcd reader. this node reads the PCD file and publishes to a desired
       topic. I have rotated the data contained in the original PCD file
       downloaded from the PCL site. In the original PCD file, all data was
       behind the sensor if we consider the frame to be the `camera_depth_optical_frame` -->
  <group ns="pcd_reader" if="$(eval source =='pcd')">
    <node name="pcd_publisher" pkg="pcl_ros" type="pcd_to_pointcloud"
          args="$(find nodelet_pcl_demo)/launch/table_rotated.pcd 0.05">
      <remap from="cloud_pcd" to="/camera/depth/points" />
      <param name="frame_id" value="camera_depth_optical_frame" />
    </node>
  </group>
  
  <node pkg="nodelet_pcl_demo" name="cluster_extraction" type="cluster_extractor" output="screen" />

  <node pkg="nodelet_pcl_demo" name="myMoveRobot_Node" type="moveRobot" output="screen" />

  <!-- start rviz if vis arg is true -->
  <node pkg="rviz" type="rviz" respawn="false" name="rviz" if="$(arg vis)"
        args="-d $(find nodelet_pcl_demo)/launch/nodelet_demo.rviz" />

</launch>
